File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        stack<char> st;
        
        int ans = 0;
        for(int i = 0;i<s.size();i++){
            if(s[i] == '('){
                st.push('(');
            }else if(s[i] == ')'){
                ans = max(ans,(int)st.size());
                st.pop();
            }
        }
        return ans;
    }
};

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.java
class Solution { // tc is O(n) and sc is O(1)
    public int maxDepth(String s) {
        int ans =0,count = 0;
        char[] str = s.toCharArray();

        for(int i = 0;i<s.length();i++){
            if(str[i] == '('){
                count++;
                ans = Math.max(ans,count);
            }else if(str[i] == ')'){
                count--;
            }
        }
        return ans;
    }
}